tests/serializers/test_data.py

```diff
@@ -429,6 +435,10 @@ def inherited_compare(testcase, pk, klass, data):
     ]
 
 
+if not connection.features.supports_index_on_text_field:
+    test_data = [data for data in test_data if data[2] != TextPKData]
```

Comment:
By my count, this makes the 3rd conditional manipulation of the `test_data` list based on `connection.features.XXX`, I feel it's starting to get a bit unwieldy.

Because the `TextPKData` model is already tagged with `Meta.required_db_features`, I wonder if there's a way to automatically skip it later in the `for func, pk, klass, datum in test_data` loop. That could be a more reusable way of skipping some tests under certain backends.

Another option would be to annotate each `test_data` element with the db feature they require, but that might require a bigger refactor of the whole test 🤔

Reply:
I agree that it's a bit unfortunate. How about I circle back to a possible refactor after all of the test classes are uncommented? That way we will know how many special cases are actually needed.

Reply:
I added a commit to show how we might address this
I agree with Baptiste that it's time for a refactor and I think this can happen sooner rather than later

The naming of stuff is not great, please take a look and let me know what you think :+1:

Reply:
I submitted a review!

Reply:
I personally really like Sarah's use of `skipUnlessDBFeature`. That's a standard way of handling this kind of situation where you want to do different things for different backends, and I'm all for using standard tools when we can.

I don't see the value of having everything inside one `test_data` list, and separating the special cases into their own lists makes sense to me.

Good work 👏🏻

Reply:
Sounds good to me!